src/microbe_stage/MicrobeHUD.cs

```diff
@@ -779,19 +783,20 @@ private void UpdateATP(float delta)
             return;
 
         var atpAmount = 0.0f;
-        var capacity = 4.0f;
 
+        // Update to the player's current HP, unless the player is dead
         if (stage.Player != null)
         {
             var compounds = GetPlayerColonyOrPlayerStorage();
 
-            atpAmount = Mathf.Ceil(compounds.GetCompoundAmount(atp));
-            capacity = compounds.Capacity;
+            // Only show ATP changes in increments of 0.5
+            atpAmount = Mathf.Ceil(compounds.GetCompoundAmount(atp) * 2) / 2.0f;
```

Comment:
What's this change? This changes the ATP display more than the title of this PR gives a clue about...

Reply:
My bad; just added it to the PR description. This allows the ATP to be displayed in half increments, since with the balance changes max ATP isn't always a whole number.

Reply:
Wouldn't it still be better to use the actual values on the bars? If needed for the numbers, ATP could be shown formatted to just one decimal place.

Reply:
The problem with the numbers is that ATP is actually changing every frame, you just don't notice when rounding since it's usually just a few decimal points at the same time. Change the 2s to 10s and you'll see the flicker.

I guess there's no problem with using the real floats for the bar though.

Reply:
Would rounding to a single decimal place be a good middle ground, still pretty good precision but it wouldn't change as rapidly as displaying the entire value?

Reply:
No, you will see a change every frame. That's what I tried first before moving to the 0.5 precision, and even then you still see it flicker once or twice sometimes when you catch it right.

Reply:
Has it always been like that, or is this a new problem that needs fixing in this PR?
@Kasterisk @buckly90 thoughts?

Reply:
My assumption here (and I didn't test it out) is that ATP cost and ATP generation are happening in different places, so sometimes the GUI thread runs after the paycheck comes in but before the bills are paid or vice versa, so sometimes the exact ATP will be a little different. If that's true, then this might be a problem with really big microbes, but I wanted to limit scope since I was already adding more to the issue with this fix.

Reply:
It does seem that after I set storage amounts to decimal numbers, the ATP bar has trouble remianing completely full when it otherwise should be. I'm not certain what exactly causes it but my assumption is it's something to do with processes shutting off when the player is at capacity, combined with low capacity and high ATP costs.

Reply:
@adQuid that's very unlikely to be the case because so far threading is only used within a single category of operations, for example compound cloud updates, AI updates, process system, meaning that the ordering between microbe _Process and the process system running (that runs the cellular processes) is fixed. What is more likely is that processes don't run if only partially their output would fit in the cell storage, to not waste resources. So the ATP amount oscillates between the ATP generation running pretty much to fill  the ATP bar, and there being a little bit empty ATP space left over, but not enough to run a process to fill up the storage.

Reply:
I guess it's fine then to do this rounding here, then.